packages/core/src/shaderlib/particle/force_over_lifetime_module.glsl

```diff
@@ -45,38 +46,50 @@
 
             if (key.x >= normalizedAge){
                 float tHat = normalizedAge * a_ShapePositionStartLifeTime.w;
-                cumulativeValue += computeDisplacementIntegral(tHat, t1, t2, lastKey.y, key.y, lastVelocity);
+                cumulativeValue += computeDisplacementIntegral(tHat, t1, t2, lastKey.y, key.y, velocityOffsetComponent);
+
+                float timeOffset = normalizedAge - lastKey.x;
+                float ratio = timeOffset / (key.x - lastKey.x);
+                float finalAcceleration = mix(lastKey.y, key.y, ratio);
+                velocityOffsetComponent += 0.5 * timeOffset * (finalAcceleration + lastKey.y);
                 break;
             } else {  
-                cumulativeValue += computeDisplacementIntegral(t2, t1, t2, lastKey.y, key.y, lastVelocity);
-                lastVelocity += 0.5 * (t2 - t1) * (lastKey.y + key.y);
+                cumulativeValue += computeDisplacementIntegral(t2, t1, t2, lastKey.y, key.y, velocityOffsetComponent);
+                velocityOffsetComponent += 0.5 * (t2 - t1) * (lastKey.y + key.y);
```

Comment:
![image](https://github.com/user-attachments/assets/98532e63-f22a-450f-8ecd-b5518d4f9035)
Are you sure, they are same uint????